addons/account/models/account_move.py

```diff
@@ -894,11 +894,55 @@ def _reconcile_lines(self, debit_moves, credit_moves, field):
             credit_move = credit_moves[0]
             company_currency = debit_move.company_id.currency_id
             # We need those temporary value otherwise the computation might be wrong below
-            temp_amount_residual = min(debit_move.amount_residual, -credit_move.amount_residual)
             temp_amount_residual_currency = min(debit_move.amount_residual_currency, -credit_move.amount_residual_currency)
             dc_vals[(debit_move.id, credit_move.id)] = (debit_move, credit_move, temp_amount_residual_currency)
             amount_reconcile = min(debit_move[field], -credit_move[field])
 
+            max_date = max(debit_move.date, credit_move.date)
+
+            # If we reconcile in amount_currency, then we need to actualize those numbers to some date
+            # Then see what is their relative value in company currency
+            if field == "amount_residual_currency":
```

Comment:
In this approach, you are not telling apart how much one amount_currency is as much the other, in case the two amount currencies are in different currencies.